Source/WebCore/Modules/mediastream/gstreamer/GStreamerRtpSenderBackend.cpp

```diff
@@ -162,17 +161,94 @@ bool GStreamerRtpSenderBackend::replaceTrack(RTCRtpSender& sender, MediaStreamTr
 
 RTCRtpSendParameters GStreamerRtpSenderBackend::getParameters() const
 {
-    return toRTCRtpSendParameters(m_initData.get());
+    switchOn(m_source, [&](const Ref<RealtimeOutgoingAudioSourceGStreamer>& source) {
+        m_currentParameters = source->parameters();
+    }, [&](const Ref<RealtimeOutgoingVideoSourceGStreamer>& source) {
+        m_currentParameters = source->parameters();
+    }, [](const std::nullptr_t&) {
+    });
+
+    GST_DEBUG_OBJECT(m_rtcSender.get(), "Current parameters: %" GST_PTR_FORMAT, m_currentParameters.get());
+    if (!m_currentParameters)
+        return toRTCRtpSendParameters(m_initData.get());
+
+    return toRTCRtpSendParameters(m_currentParameters.get());
+}
+
+static bool validateModifiedParameters(const RTCRtpSendParameters& newParameters, const RTCRtpSendParameters& oldParameters)
+{
+    if (oldParameters.transactionId != newParameters.transactionId)
+        return false;
+
+    if (oldParameters.encodings.size() != newParameters.encodings.size())
+        return false;
+
+    for (size_t i = 0; i < oldParameters.encodings.size(); ++i) {
+        if (oldParameters.encodings[i].rid != newParameters.encodings[i].rid)
+            return false;
+    }
+
+    if (oldParameters.headerExtensions.size() != newParameters.headerExtensions.size())
+        return false;
+
+    for (size_t i = 0; i < oldParameters.headerExtensions.size(); ++i) {
+        const auto& oldExtension = oldParameters.headerExtensions[i];
+        const auto& newExtension = newParameters.headerExtensions[i];
+        if (oldExtension.uri != newExtension.uri || oldExtension.id != newExtension.id)
+            return false;
+    }
+
+    if (oldParameters.rtcp.cname != newParameters.rtcp.cname)
+        return false;
+
+    if (!!oldParameters.rtcp.reducedSize != !!newParameters.rtcp.reducedSize)
+        return false;
+
+    if (oldParameters.rtcp.reducedSize && *oldParameters.rtcp.reducedSize != *newParameters.rtcp.reducedSize)
```

Comment:
Wouldn't we need to check for newParameters.rtcp.reducedSize to be valid?

Reply:
Aren't we doing it here?
BTW this function is verbatim-copied from the libwebrtc backend.

Reply:
Yes, I was reading the lines separately so I missed the check one line above. Sorry for the noise